src/FSharp.Stats/DistanceMetrics.fs

```diff
@@ -86,8 +86,82 @@ module DistanceMetrics =
             for i in 0 .. (dim - 1) do
                 let x = v1.[i] - v2.[i]
                 if not (isNan x) then
-                    dist <- dist + System.Math.Abs x
+                    dist <- dist + abs x
             dist
+
+        /// <summary>The [Minkowski distance](https://en.wikipedia.org/wiki/Minkowski_distance) between two vectors of order `p`.</summary>
+        /// <remarks>The two vectors need not have equal lengths: when one vectors is exhausted any remaining elements in the other vectors are ignored.</remarks>
+        /// <param name="s1">first vector</param>
+        /// <param name="s2">second vector</param>
+        /// <param name="p">float constrained to `p > 0`</param>
+        /// <returns>Minkowski distance between elements of given vectors. Returns NaN if vectors contain NaN.</returns>
+        /// <example> 
+        /// <code> 
+        /// // e.g. v1 and v2 initialization
+        /// let v1 = vector [3.14; 2.0; 3.1]
+        /// let v2 = vector [9.1; 2.5; 3.7]
+        /// 
+        /// // Apply the minkowski distance to v1 and v2
+        /// Vector.minkowski v1 v2 3
+        /// </code> 
+        /// </example>
+        let inline minkowski (v1: Vector<'a>) (v2: Vector<'a>) (p: float) : float option =
+            if p <= 0.0 then
+                None
+            else
+                let dim = min v1.Length v2.Length
+                let mutable dist = 0.0
+
+                for i in 0 .. (dim - 1) do
+                    let diff = 
```

Comment:
Sorry, my previous comment may was unclear. I wondered why instead of using `abs` you defined your own `abs difference` function

```fsharp
let diff =
    if v1.[i] > v2.[i] then
        v1.[i] - v2.[i] 
    else
        v2.[i] - v1.[i]
```

This could be changed to `abs (v1.[i] - v2.[i])`

Reply:
you can also find this logic in the `cityblock` function versions with the generic inputs.
If you use an  `abs` here the compiler casts it to a float.